### ISSUE TITLE
ttnn.logit produces incorrect results for eps > 0.5, masked by platform-dependent UB in test golden

### ISSUE THREAD

**Issue by mcraigheadTT** [tenstorrent/tt-metal#41537]
## Summary

`ttnn.logit(x, eps=...)` produces incorrect results when `eps > 0.5`. The kernel outputs a step function (constant +/- values) instead of actual logit values. This bug is masked on x86_64 because `torch.special.logit` has its own undefined behavior for `eps > 0.5` that coincidentally produces the same wrong answer, causing the test to pass.

The bug was discovered while running tests on aarch64 (Ubuntu 24.04), where `torch.special.logit`'s UB resolves differently, exposing the mismatch.

## Affected test

`tests/ttnn/unit_tests/operations/eltwise/test_unary.py::test_unary_logit` -- all parameter combinations with `scalar=0.85` fail on aarch64, but the underlying kernel bug affects all `eps > 0.5` values on all platforms.

## Root cause

There are two independent bugs:

### Bug 1: Kernel produces wrong results for eps > 0.5

**File:** `ttnn/cpp/ttnn/operations/eltwise/unary/device/kernels/compute/logit_kernel.cpp`

When `eps > 0.5`, the host-side program factory (`unary_ng_program_factory.cpp:51-63` and `unary_program_factory.cpp:137-148`) passes the kernel:
- `packed_scalar1 = eps` (e.g. 0.85)
- `packed_scalar2 = 1 - eps` (e.g. 0.15)

The kernel calls `clamp_tile(0, packed_scalar1, packed_scalar2)` which clamps to `[0.85, 0.15]` -- inverted bounds. This degenerately clamps **all** values to `1-eps` (0.15). Then:
1. `logit(0.15) = log(0.15/0.85) = -1.735` for every element
2. The `#ifdef WHERE` block conditionally negates based on `original_input < eps`
3. Result: a step function -- `+1.735` for all `x < eps`, `-1.735` for all `x >= eps`

This is **not** the correct logit function. Per the [PyTorch docs](https://pytorch.org/docs/stable/special.html#torch.special.logit), the piecewise definition for `eps > 0.5` should clamp `x < eps` to `eps` and `x > 1-eps` to `1-eps`, giving:
- `x = 0.20` -> clamped to `0.20` (within `[0.15, 0.85]`) -> `logit(0.20) = -1.386`
- `x = 0.50` -> clamped to `0.50` -> `logit(0.50) = 0.0`
- `x = 0.85` -> clamped to `0.85` -> `logit(0.85) = +1.735`

But the kernel returns `+1.735` for all three of these.

**Suggested fix:** Replace `clamp_tile(0, packed_scalar1, packed_scalar2)` with properly ordered bounds: `clamp_tile(0, min(eps, 1-eps), max(eps, 1-eps))`. Then the WHERE hack for conditional negation should no longer be needed and can be removed.

### Bug 2: Test golden function has platform-dependent undefined behavior

**File:** `ttnn/ttnn/operations/unary.py:585-588`

```python
def _golden_function_logit(input_tensor_a, *args, eps=None, **kwargs):
    import torch
    return torch.special.logit(input_tensor_a, eps=eps)
```

`torch.special.logit` internally calls the equivalent of `std::clamp(x, eps, 1-eps)`. When `eps > 0.5`, this is `std::clamp(x, 0.85, 0.15)` with `lo > hi`, which is **undefined behavior** per the C++ standard. Different platforms resolve the UB differently:

| Platform | `torch.special.logit(0.5, eps=0.85)` | Likely `std::clamp` impl |
|----------|--------------------------------------|--------------------------|
| x86_64 (Ubuntu 22.04) | `+1.735` | `x < lo ? lo : x > hi ? hi : x` |
| aarch64 (Ubuntu 24.04) | `-1.735` | `min(hi, max(x, lo))` |

On x86_64, torch's UB coincidentally produces the same step function as the buggy kernel, so tests pass. On aarch64, torch produces a different (also wrong) constant, so tests fail.

**Suggested fix:** Either implement the golden without `torch.special.logit` for `eps > 0.5`:
```python
def _golden_function_logit(input_tensor_a, *args, eps=None, **kwargs):
    import torch
    if eps is not None:
        lo = min(eps, 1.0 - eps)
        hi = max(eps, 1.0 - eps)
        x = torch.clamp(input_tensor_a, lo, hi)
        return torch.special.logit(x)  # eps=None, already clamped
    return torch.special.logit(input_tensor_a)
```
Or document that `eps > 0.5` is unsupported and remove those test parameters.

### Kernel output is identical on both platforms

The monolithic `ttnn.logit` kernel produces the same output on x86_64 and aarch64:

```
x= -1.0000  monolithic= +1.734601
x= +0.5000  monolithic= +1.734601
x= +0.8500  monolithic= -1.734601
x= +1.0000  monolithic= -1.734601
```

### Golden output differs across platforms

```
                  aarch64 golden    x86_64 golden
x= -1.0000       -1.734601         +1.734601
x= +0.5000       -1.734601         +1.734601
x= +0.8500       -1.734601         -1.734601
x= +1.0000       -1.734601         -1.734601
```

### Manual decomposition produces correct results

Breaking logit into individual ttnn ops (`clip` with properly ordered bounds -> `subtract` -> `divide` -> `log`) produces correct results on all platforms, confirming the individual SFPU operations are fine. The bug is specifically in the fused logit kernel's handling of inverted clamp bounds + WHERE conditional negation.

### All eps > 0.5 values are affected

Tested on aarch64 (where the golden divergence exposes the issue):

| eps | ATOL | RTOL | Status |
|-----|------|------|--------|
| 0.25 | 0.0 | 0.0 | PASS |
| 0.38 | 0.0 | 0.0 | PASS |
| 0.50 | 0.0 | 0.0 | PASS |
| 0.51 | 0.08 | 2.0 | FAIL |
| 0.55 | 0.40 | 2.0 | FAIL |
| 0.60 | 0.81 | 2.0 | FAIL |
| 0.85 | 3.47 | 2.0 | FAIL |

Note the RTOL is exactly 2.0 in all failing cases -- a sign flip.

## Reproduction

Diagnostic test (run with `pytest -s`):

```python
import torch
import ttnn

def test_logit_eps_above_half(device):
    vals = [-1.0, 0.0, 0.15, 0.5, 0.85, 1.0]
    padded = torch.zeros(1, 1, 32, 32, dtype=torch.float32)
    padded.flatten()[:len(vals)] = torch.tensor(vals)

    inp = ttnn.from_torch(padded, dtype=ttnn.float32, layout=ttnn.TILE_LAYOUT, device=device)
    result = ttnn.to_torch(ttnn.logit(inp, eps=0.85))

    # What the kernel produces (same on all platforms):
    # x < 0.85: +1.734601 (wrong -- should vary per input)
    # x >= 0.85: -1.734601
    for i, x in enumerate(vals):
        print(f"x={x:+.2f}  result={result.flatten()[i].item():+.6f}")
```

## Environment

- Discovered on: aarch64, Ubuntu 24.04, ttsim (Wormhole simulator)
- Confirmed kernel output identical on: x86_64, Ubuntu 22.04, ttsim and silicon
- Relevant kernel files:
  - `ttnn/cpp/ttnn/operations/eltwise/unary/device/kernels/compute/logit_kernel.cpp` (lines 44-88)
  - `ttnn/cpp/ttnn/operations/eltwise/unary_ng/device/unary_ng_program_factory.cpp` (lines 51-63)
  - `ttnn/cpp/ttnn/operations/eltwise/unary/device/unary_program_factory.cpp` (lines 137-148)
  - `ttnn/ttnn/operations/unary.py` (lines 585-588)
